sdk/src/logs/logger.cc

```diff
@@ -36,23 +51,76 @@ void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
 
   // TODO: Sampler logic (should include check for minSeverity)
 
-  /**
-   * Convert the LogRecord to the heap first before sending to processor.
-   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
-   * converting record a heap variable can be removed
-   */
-  auto record_pointer =
-      std::unique_ptr<opentelemetry::logs::LogRecord>(new opentelemetry::logs::LogRecord(record));
+  auto recordable = processor->MakeRecordable();
+  if (recordable == nullptr)
+  {
+    return;
+  }
+
+  // Populate recordable fields
+  if (timestamp == this->default_timestamp_)
+  {
+    // Inject current timestamp if none is specified by user
+    recordable->SetTimestamp(core::SystemTimestamp(std::chrono::system_clock::now()));
+  }
+  else
+  {
+    recordable->SetTimestamp(timestamp);
+  }
+
+  recordable->SetSeverity(severity);
+  recordable->SetName(name);
+  recordable->SetBody(body);
+
+  resource.ForEachKeyValue([&](nostd::string_view key,
+                               opentelemetry::common::AttributeValue value) noexcept {
+    recordable->SetResource(key, value);
+    return true;
+  });
+
+  attributes.ForEachKeyValue([&](nostd::string_view key,
+                                 opentelemetry::common::AttributeValue value) noexcept {
+    recordable->SetAttribute(key, value);
+    return true;
+  });
+
+  // Inject trace_id/span_id/trace_flags if none is set by user
+  auto provider     = opentelemetry::trace::Provider::GetTracerProvider();
+  auto tracer       = provider->GetTracer("foo_library");
```

Comment:
What is "foo_library", default tracer name?

Reply:
I didn't notice I left this in - from an earlier example written. Yes, it seems `GetTracer()` doesn't accept an empty param list? Would it be better to remove injection from this PR completely, as it seems to be a separate topic? Or is there a better default name I could use here (which I could also revisit the naming in a later PR)?

Reply:
Fixed - added Logger name and updated the Tracer name above to be the same in latest commit